core/variable.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2012 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Components for the variable model.
+ * @author marisaleung@google.com (Marisa Leung)
+ */
+'use strict';
+
+goog.provide('Blockly.Variable');
+
+/**
+ * Class for a variable.
+ * @param {!string} name The name of the variable.
+ * @param {!string} type The type of the variable.
+ * @param {!string} id The unique id of the variable.
+ * @constructor
+ */
+Blockly.Variable = function(name, type, id) {
+  /*
```

Comment:
The comment style I used in that doc is magically not the same as our normal comment style.

Follow this example:
https://github.com/google/blockly/blob/master/core/block_svg.js#L86

So for a private variable:
```
/**
 * Description.
 * @type {type}
 * @private
 */
```

And a public one:
```
/**
 * Description.
 * @type {type}
 */
```